tools/windows-node-installer/cmd/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+	logger "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	// log is the global logger for the main package.
+	log = logger.Log.WithName("windows-node-installer")
+
+	// rootInfo contains necessary information for creating Cloud instance from cloudprovider package.
+	rootInfo struct {
+		kubeconfigPath      string
+		credentialPath      string
+		credentialAccountID string
+		resourceTrackerDir  string
+	}
+)
+
+// main is the entry point for windows-node-installer.
+func main() {
+	rootCmd := newRootCmd()
+
+	for _, subCmd := range []*cobra.Command{
+		createCmd(),
+		destroyCmd(),
+	} {
+		rootCmd.AddCommand(subCmd)
+	}
+	err := rootCmd.Execute()
+	if err != nil {
+		log.Error(err, "error executing window-node-installer")
+	}
+}
+
+// newRootCmd defines `windows-node-installer` command with persistent flags to fill up information in rootInfo.
+// It uses PreRunE to check for whether required flags are provided.
+func newRootCmd() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:   "windows-node-installer",
+		Short: "Creates and destroys Windows instance with an existing OpenShift cluster.",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return requiredRootFlags(cmd)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return cmd.Help()
+		},
+	}
+
+	rootCmd.PersistentFlags().StringVar(&rootInfo.kubeconfigPath, "kubeconfig", "",
+		"file path to the kubeconfig of the existing OpenShift cluster.")
+
+	rootCmd.PersistentFlags().StringVar(&rootInfo.credentialPath, "credentials-dir", "",
```

Comment:
Please change this from `credentials-dir`, to `credentials`, what you have currently makes it sound like you want the path to the `~/.aws` directory